### PR TITLE
Change to BLE low level API

### PR DESCRIPTION
**Pull request by huguesBouvier**
Change to BLE low level API

Description
-----------
Change to BLE low level API
- Adding WhiteListing
- Added stop advertisement callback
- Fixed test and middlware for the new APIs
- Fixed forgotten BaseType in middleware
- Improved comment in API

Test: 
- Demo and test for ESP (100% PASS)
- test for Nordic (100% PASS)

Checklist:
----------

- [c ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/ble_hal/include/bt_hal_gatt_server.h [hunk 1]
```diff
@@ -687,6 +698,72 @@ typedef struct
...
+     * Triggers BTWhiteListChangedCallback_t with bIsAdded = true.
+     *
+     * @param[in] ucServerIf Server interface, return on the callback BTRegisterServerCallback_t
+     *  after successful pxRegisterServer call.
+     *
+     * @param[in] pxBdAddr Address of the Remote device.
+     *
+     * @param[in] ucNumberOfDevices Number of bluetooth device address passes
+     *
+     * @param[in] xTransport Specify if BLE and BT classic is being used.
```
Comment (ravibhagavandas) [inline libraries/abstractions/ble_hal/include/bt_hal_gatt_server.h]: Unused parameter ?

File: libraries/abstractions/ble_hal/include/bt_hal_gatt_server.h [hunk 2]
```diff
@@ -687,6 +698,72 @@ typedef struct
...
+
+    /**
+     * @brief Add devices to whitelist
+     *
+     * Triggers BTWhiteListChangedCallback_t with bIsAdded = true.
+     *
+     * @param[in] ucServerIf Server interface, return on the callback BTRegisterServerCallback_t
+     *  after successful pxRegisterServer call.
+     *
+     * @param[in] pxBdAddr Address of the Remote device.
```
Comment (ravibhagavandas) [inline libraries/abstractions/ble_hal/include/bt_hal_gatt_server.h]: Is an array of device addresses passed in ? If so can you update this doc accordingly ?

File: libraries/abstractions/ble_hal/include/bt_hal_gatt_server.h (hunk 1 above)
Comment (huguesBouvier) [reply libraries/abstractions/ble_hal/include/bt_hal_gatt_server.h]: Fixed

File: libraries/abstractions/ble_hal/include/bt_hal_gatt_server.h (hunk 2 above)
Comment (huguesBouvier) [reply libraries/abstractions/ble_hal/include/bt_hal_gatt_server.h]: fixed

**Pull request merged**